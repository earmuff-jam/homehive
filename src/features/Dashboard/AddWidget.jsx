import React from "react";

import {
  Divider,
  ListItemIcon,
  ListItemText,
  MenuItem,
  MenuList,
  Stack,
  Typography,
} from "@mui/material";
import { AddRounded } from "@mui/icons-material";

const WidgetTypeList = [
  {
    id: 1,
    label: "Details Chart",
    inset: false, // makes text have extra spacing infront
  },
  {
    id: 2,
    label: "Invoice Details",
    inset: false,
  },
  {
    id: 3,
    label: "Invoice Demographics",
    inset: false,
  },
];

export default function AddWidget() {
  return (
    <Stack sx={{ minWidth: "12rem", padding: "1rem" }}>
      <Typography> Add Widget</Typography>
      <Divider />
      <MenuList dense>
        {WidgetTypeList.map((widgetType) => (
          <MenuItem key={widgetType.id}>
            <ListItemIcon>
              <AddRounded />
            </ListItemIcon>
            <ListItemText inset={widgetType.inset}>
              {widgetType.label}
            </ListItemText>
          </MenuItem>
        ))}
      </MenuList>
    </Stack>
  );
}
