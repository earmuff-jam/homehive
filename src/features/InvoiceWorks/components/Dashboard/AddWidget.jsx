import React from "react";

import { AddRounded } from "@mui/icons-material";
import {
  Divider,
  ListItemIcon,
  ListItemText,
  MenuItem,
  MenuList,
  Stack,
  Typography,
} from "@mui/material";
import { WidgetTypeList } from "features/InvoiceWorks/components/Dashboard/constants";

export default function AddWidget({ handleAddWidget }) {
  return (
    <Stack sx={{ minWidth: "12rem", padding: "1rem" }}>
      <Typography> Add Widget</Typography>
      <Divider />
      <MenuList dense>
        {WidgetTypeList.map((widgetType) => (
          <MenuItem
            key={widgetType.id}
            onClick={() => handleAddWidget(widgetType.id)}
          >
            <ListItemIcon>
              <AddRounded />
            </ListItemIcon>
            <ListItemText inset={widgetType.config.inset}>
              {widgetType.label}
            </ListItemText>
          </MenuItem>
        ))}
      </MenuList>
    </Stack>
  );
}
