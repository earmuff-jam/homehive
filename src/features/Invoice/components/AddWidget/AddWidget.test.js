import React from "react";

import AddWidget from "./AddWidget";
import { fireEvent, render, screen } from "@testing-library/react";

jest.mock("features/Invoice/constants", () => ({
  WidgetTypeList: [
    { id: "chart", label: "Chart", config: {} },
    { id: "summary", label: "Summary", config: {} },
  ],
}));

describe("AddWidget Component", () => {
  describe("AddWidget Component snapshot tests", () => {
    test("Footer matches snapshot", () => {
      const { asFragment } = render(<AddWidget />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe("AddWidget Component tests", () => {
    it("renders the title and widget list", () => {
      render(<AddWidget handleAddWidget={jest.fn()} />);

      expect(screen.getByText("Add Widget")).toBeInTheDocument();

      expect(screen.getByText("Chart")).toBeInTheDocument();
      expect(screen.getByText("Summary")).toBeInTheDocument();
    });

    it("calls handleAddWidget with correct id on click", () => {
      const handleAddWidget = jest.fn();
      render(<AddWidget handleAddWidget={handleAddWidget} />);

      fireEvent.click(screen.getByText("Chart"));

      expect(handleAddWidget).toHaveBeenCalledTimes(1);
      expect(handleAddWidget).toHaveBeenCalledWith("chart");
    });
  });
});
