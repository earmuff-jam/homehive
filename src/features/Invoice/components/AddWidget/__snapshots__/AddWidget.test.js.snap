// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWidget Component AddWidget Component snapshot tests Footer matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-1aczxsw-MuiStack-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
       Add Widget
    </p>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
    />
    <ul
      class="MuiList-root MuiList-padding MuiList-dense css-b7dj7n-MuiList-root"
      role="menu"
      tabindex="-1"
    >
      <li
        class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1ugarj1-MuiButtonBase-root-MuiMenuItem-root"
        role="menuitem"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cokf1l-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="AddRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-cfq8qh-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
          >
            Chart
          </span>
        </div>
      </li>
      <li
        class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-1ugarj1-MuiButtonBase-root-MuiMenuItem-root"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="MuiListItemIcon-root css-cokf1l-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="AddRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense css-cfq8qh-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-7ctihp-MuiTypography-root"
          >
            Summary
          </span>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
